hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/monitoring/MonitoringCache.java

```diff
@@ -0,0 +1,164 @@
+package pl.allegro.tech.hermes.management.infrastructure.monitoring;
+
+import com.google.common.collect.Lists;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.common.kafka.KafkaNamesMapper;
+import pl.allegro.tech.hermes.management.config.MonitoringProperties;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaClustersProperties;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaNamesMappers;
+import pl.allegro.tech.hermes.management.config.kafka.KafkaProperties;
+import pl.allegro.tech.hermes.management.domain.subscription.SubscriptionService;
+import pl.allegro.tech.hermes.management.domain.topic.TopicService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL;
+import static org.apache.kafka.clients.CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.SECURITY_PROTOCOL_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_JAAS_CONFIG;
+import static org.apache.kafka.common.config.SaslConfigs.SASL_MECHANISM;
+import static pl.allegro.tech.hermes.api.TopicName.fromQualifiedName;
+
+public class MonitoringCache {
+
+    private static final Logger logger = LoggerFactory.getLogger(MonitoringCache.class);
+
+    private final KafkaClustersProperties kafkaClustersProperties;
+    private final KafkaNamesMappers kafkaNamesMappers;
+    private final MonitoringProperties monitoringProperties;
+    private final SubscriptionService subscriptionService;
+    private final TopicService topicService;
+
+    private volatile List<List<TopicAndSubscription>> readySubscriptionsWithUnassignedPartitionsCache = new ArrayList<>();
+    private volatile List<List<TopicAndSubscription>> unreadySubscriptionsWithUnassignedPartitionsCache = new ArrayList<>();
```

Comment:
You propably don't need this variable at all to be at the object level. It is quite complicated now. Maybe better would be:
1. preserve `readySubscriptionsWithUnassignedPartitionsCache` as it is
2. remove `unreadySubscriptionsWithUnassignedPartitionsCache` from the object itself and move the creation of this list to the method responsible for fetching kafka consumer groups
3. once a fetching is done you just switch `readySubscriptionsWithUnassignedPartitionsCache` to the new collection created in point 2.